### ISSUE TITLE
SD Card Corruption

### ISSUE THREAD

**Issue by ToxiKutie** [d0k3/GodMode9#135]
I'm back again with a much more serious issue. I'm not sure if this has happened before, and I searched for a similar case, but got nothing. I eject my SD card safely every time, so I know it wasn't my Computer. What you can see showed up after I returned the card to my 3DS. The CIA files I moved remained intact, strangely enough.

![snap000](https://user-images.githubusercontent.com/29241060/28751481-9a1e4bcc-74bd-11e7-9a93-f80d48f4fac8.png)

**Comment by d0k3** [comment]
Check that card in a PC (also check filesystem integrity with chkdsk or similar). What do you get?

**Comment by d0k3** [comment]
@ToxiKutie - this misses info. The problem is most likely with your SD card, not GM9.

**Comment by ToxiKutie** [comment]
I checked the SD card in my PC afterward and it produced the same result, including the error messages stating the filesystem was corrupt and inaccessible. I used diskpart after to check the emunand partition, which remained fully intact and functional, which I am using as I write this. The main partition formatted fine after this, and I luckily had nothing on it as I had just formatted the EmuNAND with TinyFormat and was setting it up when the corruption hit. The only piece of damning evidence I can provide is that I left the SD card unmounted for an unexpectedly long time with EmuNAND level 1 write permissions unlocked.

**Comment by d0k3** [comment]
It's doubtable GM9 is at fault here, but let's sort this out...

Just a quick note, the kind of corruption you got there is not one that would happen when unsafely ejecting your SD.

Did you write anything to your EmuNAND before it happened? Also, is your EmuNAND in any way "special"? Is this a GW type EmuNAND, or is it a RedNAND? What type of console is this and what did you use to create the EmuNAND? Is this a multi EmuNAND?

The only entry in there that I recognize is NCSD, and that looks an awful lot like your root directory got overwritten with an NCSD header - which is in itself very odd.

**Comment by ToxiKutie** [comment]
EmuNAND GW-type, Single, Made with GM9 itself, and on an O3DS. I had not written anything to it, simply erased my NNID save from it for a fresh start. 

I had a slight feeling that it wasn't related to the ejection itself, but Any detail is a good one, yes?

**Comment by d0k3** [comment]
You didn't do any NCSD header edits (NAND repartitioning , right?). What is the size of your EmuNAND (hint: it's the size of nand.bin in E:)

**Comment by ToxiKutie** [comment]
The partition size I always use for my EmuNAND is the Legacy 2GB, and the size of the nand.bin itself is 943MB. I did not perform any type of repartioning beyond the initial partitioning through GM9. This particular SD card is either in my computer via the SD reader or in my 3DS with nothing in-between. The header writeout only became visible once I returned the SD Card to the DS itself, however, but my computer has no way of reading the NCSD info given it's not exactly dumped to it anywhere.

**Comment by ToxiKutie** [comment]
I just remembered something else that I had done before the "corruption". I mounted a .cia file and injected the .app into my EmuNAND H&S. Though that doesn't seem relatively close to what caused this (Especially given it generated two files over 3GB in size) it seems worthy to note that it was after that, and leaving it out for the extended period of time was what took place before this.

_Sent from my ZTE Z799VL using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by d0k3** [comment]
Well, the size you see there, in the file listing is not correct. If that inject actually worked like it should in your EmuNAND, that is also not the cause of the corruption.

**Comment by ToxiKutie** [comment]
Oh, it was after. I just thought it should be mentioned that I mounted said image before this happened.

**Comment by MelonGx** [comment]
BTW, you should have NO Chinese, NO Japanese, NO Korean in your file/folder name, didn't you?

**Comment by ToxiKutie** [comment]
@MelonGx I did not, as you can see from the screenshot. None of the directories were accessible regardless.

_Sent from my ZTE Z799VL using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by d0k3** [comment]
Thanks @MelonGx. I tried to get behind this, also talking that issue through with other people, and I still have no clue. @ToxiKutie - you're the only person with that issue so far, so this very much looks like random SD corruption, which could have happened anywhere.

I suggest we will keep this issue open for another few days, hoping that someone else can add useful info. Also, @ToxiKutie - keep that SD card under observation. Maybe something is actually wrong with it.

**Comment by ToxiKutie** [comment]
@d0k3, I most definitely will. If I didn't have a synced Pokémon Sun installed to my SD Card along with (several important) other programs, I would attempt to recreate it. However, if I can get a hold of a fresh SD Card, I will do what I can to see if I can cause this again.

_Sent from my ZTE Z799VL using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by NinjaBoyLao** [comment]
I have several SD cards that I know are fine. Can either of you give me a specific set of things to try to attempt to reproduce this?

**Comment by ToxiKutie** [comment]
@NinjaBoyLao The way I caused this was with a 16gb card specifically, (not that that might matter.) Set up the EmuNand with 2gb (Legacy Size) and make it a GW type. Then unmount the card and put it into your computer, and move a few small files to it, and maybe the Luma or other folders. Now here's the thing, I accidentally left it out for roughly half an hour before returning it to the device (with the device still on GM9). 

_Sent from my ZTE Z799VL using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by d0k3** [comment]
Leaving it out that long can't have anything to do with it. Also, it's not unlikely the corruption happened on your computer. I still hope you can recreate it, though.

**Comment by d0k3** [comment]
@NinjaBoyLao - can you perhaps try to reproduce the steps @ToxiKutie described?

**Comment by d0k3** [comment]
Okay, after a thorough check of the source code, also considering how that issue happened (only showed up after the SD was in a PC, and then back in the 3DS console), also this issue being an isolated case, I'm almost 100% sure GM9 can't be the culprit. SD cards do strange stuff sometimes, and here the cause msot likely lies elsewhere.

For this reason I'm closing this now. @ToxiKutie - if you can bring any more info on this, if you can reproduce it or show us that GM9 was at fault, please feel free to reopen.

**Comment by NinjaBoyLao** [comment]
Several months after school started back and getting swamped by classes again, I'm here. I'm really sorry to be late. I followed the steps @ToxiKutie described. I also played around with a few smallish cards, a few 4GBs and a 16GB as described. the 4GBs are different brands, one is a sony and the other is a sandisk, the 16 is a samsung microsd card in an adaptor. I can't get anything strange and I even tried "unsafely" removing them from both the PC and 3DS several times.

Again, sincere apologies for being so unresponsive for so long, I'm on fall break right now and I'm catching up on all the stuff that got forcefully dropped in favor of classes.
